core/src/main/java/org/apache/gravitino/listener/FilesetEventDispatcher.java

```diff
@@ -138,4 +141,22 @@ public boolean dropFileset(NameIdentifier ident) {
       throw e;
     }
   }
+
+  @Override
+  public String getFileLocation(NameIdentifier ident, String subPath)
+      throws NoSuchFilesetException {
+    try {
+      String actualFileLocation = dispatcher.getFileLocation(ident, subPath);
+      // get the audit info from the thread local context
+      CallerContext context = CallerContext.CallerContextHolder.get();
```

Comment:
Here we can get the thread local audit context, and put it into the event bus to audit.